pkg/cloud/ovirt/clients/machineservice.go

```diff
@@ -323,8 +323,51 @@ func (is *InstanceService) handleNics(vmService *ovirtsdk.VmService, spec *ovirt
 	return nil
 }
 
+func (is *InstanceService) getAffinityGroups(cID string, agNames []string) (ag []*ovirtsdk.AffinityGroup, err error) {
+	var ags []*ovirtsdk.AffinityGroup
+	res, err := is.Connection.SystemService().ClustersService().
+		ClusterService(cID).AffinityGroupsService().
+		List().Send()
+	if err != nil {
+		return nil, err
+	}
+	agNamesMap := make(map[string]*ovirtsdk.AffinityGroup)
+	for _, af := range res.MustGroups().Slice() {
+		agNamesMap[af.MustName()] = af
+	}
+	for _, agName := range agNames {
+		if _, ok := agNamesMap[agName]; ok {
+			ags = append(ags, agNamesMap[agName])
+		}
+	}
+	return ags, nil
+}
+
+func (is *InstanceService) handleAffinityGroups(vm *ovirtsdk.Vm, cID string, agsName []string) error {
+	ags, err := is.getAffinityGroups(cID, agsName)
+	if err != nil {
+		return err
+	}
+	agService := is.Connection.SystemService().ClustersService().
+		ClusterService(cID).AffinityGroupsService()
+	for _, ag := range ags {
+		klog.Infof("Adding machine %v to affinity group %v", vm.MustName(), ag.MustName())
+		_, err = agService.GroupService(ag.MustId()).VmsService().Add().Vm(vm).Send()
+
+		// TODO: bug 1932320: Remove error handling workaround when BZ#1931932 is resolved and backported
+		if err != nil && errors.Is(err, ovirtsdk.XMLTagNotMatchError{}) {
```

Comment:
should it be `!errors.Is(...)`?

Reply:
You are right and I think that the fact that I didn't return on my testing means there is a bigger problem with the logic here